query_builder/advanced.py

```diff
@@ -176,11 +196,13 @@ def get_filter_query(self, table, query, config):
     def get_having_query(self, table, query, config):
         if self.andKey in config:
             rootConfig = config.get(self.andKey)
+            # a way to handle complex having configs
+            crit = Criterion.all([ self.get_having_criterion(table, config) for config in rootConfig ])
         elif self.orKey in config:
             rootConfig = config.get(self.orKey)
+            # a way to handle complex having configs
+            crit = Criterion.all([ self.get_having_criterion(table, config) for config in rootConfig ])
 
-        # a way to handle complex having configs
-        crit = Criterion.all([ self.get_having_criterion(table, config) for config in rootConfig ])
```

Comment:
Why did u duplicate this on line 199 and 203 when having it here is exactly the same thing, only without the duplication? Or do you expect a scenario where there's neither AND/OR? hmmm... okay - handle it properly then.

Reply:
This is what you were supposed to change, instead you changed the `get_filter_query`. Please revert that ... as for this bit, leave it as it but change the Criterion on line 203 from `Criterion.all` to `Criterion.any` ...